### PR TITLE
Cherry-pick changes from mayastor PR

### PR DESCRIPTION
**Pull request by tiagolobocastro**
Cherry-pick user changes from stale mayastor PR.
Modified the initContainers change as we now have many init containers.

### CODE REVIEW

**Review by niladrih — COMMENTED** [review]
The if condition here should also depend on `etcd.enabled`
https://github.com/openebs/mayastor-extensions/blob/1609f7a779426abc63b5015ad5fb9086003788e6/chart/templates/etcd/storage/localpv-storageclass.yaml

File: chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -287,4 +287,39 @@ Get the Jaeger URL
...
+{{- define "render_init_containers" -}}
+    {{- $containers := list }}
+    {{- $image := .Values.base.initContainers.image }}
+    {{- range .value -}}
+        {{ $container := . }}
+        {{- if not (hasKey . "image") }}
+            {{- with $image }}
+                {{- $_ := set $container "image" . }}
+            {{- else }}
+                {{- $_ := set $container "image" "docker.io/bitnami/bitnami-shell:10" }}
```
Comment (niladrih) [inline chart/templates/_helpers.tpl]: Should the default image be configurable via values.yaml? This could pose a problem for air-gapped folks.

Comment (tiagolobocastro) [reply chart/templates/_helpers.tpl]: hmm perhaps you misunderstood the code? the default is in the values.yaml
If not present, ex: upgrades with --reuse then we hardcode it here.

Comment (niladrih) [reply chart/templates/_helpers.tpl]: Oh, that's fair.

File: chart/values.yaml [hunk 2]
```diff
@@ -578,8 +576,8 @@ loki-stack:
       runAsUser: 1001
     # initContainers to chown the static hostpath PV by 1001 user
     initContainers:
-      - command: ["/bin/bash", "-ec", "chown -R 1001:1001 /data"]
-        image: docker.io/bitnami/bitnami-shell:10
+      - command: ["/bin/sh", "-ec", "chown -R 1001:1001 /data"]
+        image: docker.io/busybox:1.36.1-musl
```
Comment (niladrih) [inline chart/values.yaml]: That's very appropriate :O

**Review by niladrih — APPROVED** [review]

**Review by Abhinandan-Purkait — APPROVED** [review]

File: chart/values.yaml (hunk 2 above)
Comment (tiagolobocastro) [reply chart/values.yaml]: though I wonder if we should put our own on openebs's dockerhub?
Sadly for etcd, their init container has the init hardcoded to use bash IIRC, so we'd still need one with bash.

Comment (niladrih) [reply chart/values.yaml]: Sure, I can see the utility in that. For someone behind an air-gap, they might be taking stock of all the containers they'd be needing.
This one already lives on the openebs org: https://github.com/openebs/linux-utils/blob/main/Dockerfile
We could add bash to the apk list. Also it's alpine, so it'd have musl. But even if it didn't, everything we build is statically built and our shell scripts don't really depend on the libc implemention differences of musl/glibc, etc.

**Review by niladrih — APPROVED** [review]

**Review by niladrih — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
> The if condition here should also depend on `etcd.enabled` https://github.com/openebs/mayastor-extensions/blob/1609f7a779426abc63b5015ad5fb9086003788e6/chart/templates/etcd/storage/localpv-storageclass.yaml

Done

**Review by niladrih — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**